### ISSUE TITLE
Stability in the results for surface reconstruction

### ISSUE THREAD

**Issue by ThibaultGROUEIX** [mabaorui/NeuralPull#7]
Hi @mabaorui,

Congrats again and sorry for the repeated queries. I am still trying to figure out a few aspects of the method and am thus opening various discussions, I hope that is ok.

I have been experimenting for surface reconstruction using the Pytorch version of the code. As you know : https://github.com/bearprin/neuralpull-pytorch/issues/2

I observe that the results are not very stable aven after a long time of overfitting a single shape. I can pick a good reconstruction by selecting a step where the fitted mesh is close to the input pointcloud with an ad-hoc metric (like L1 or L2 or Hausdorff), but the next step might be a poor reconstruction.

I wonder if this instability is a bug or a normal behaviour of the method?

Thanks a lot,
Thibault
